### ISSUE TITLE
spotlessKotlin Gradle task appears to be blowing away generated code

### ISSUE THREAD

**Issue by andrewparmet** [diffplug/spotless#670]
I'm using the OpenAPI generator to generate some Kotlin code to `build/generated/openapi`. When running `spotlessKotlin` with `--info`, I see: 

```
Deleting stale output file: /path-to-project/build/generated/openapi/src/main/kotlin/com/package/GeneratedApiModel.kt`.
```

I'm declaring that path to be a source directory:
```kotlin
sourceSets["main"].java {
    srcDir("$buildDir/generated/openapi/src/main/kotlin")
}
```

And I'm not doing any strange Gradle trickery to that directory otherwise.

From my Googling it looks like `Deleting stale output file` is related to task output declaration, but I'm not sure how Gradle would come to understand this directory as an output of `spotlessKotlin`. I can work around the issue by `targetExclude`ing that directory (which I actually want to do anyways, as I don't want to lint the generated code).

Basically I'm wondering why this happens? It's odd to me that the Spotless Gradle plugin seems to be deleting code.

**Comment by nedtwigg** [comment]
What version of Spotless are you using?  We've been doing lazy-config incremental up-to-date checking since Gradle 2.x, and that required bashing Gradle's guts a little bit, which Gradle got less and less happy with as new versions of Gradle came out.

In Spotless [5.0](https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md#500---2020-07-12) we bumped our minimum required Gradle from `2.x` to `5.4`, and we now do everything "kosher".  The only output directory we declare is this one:

https://github.com/diffplug/spotless/blob/de2d84ef15a4295ff6ed4ed86377d25669cb1e6a/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java#L138-L143

so it shouldn't be causing the problem you are seeing.  In older versions of Spotless, we declared the source directories as both input and output, which is a little circular, but it let us make incremental up-to-date work (with a little trickery) on the old 2.x APIs.

**Comment by andrewparmet** [comment]
We were using 3.28.1. Upon upgrading to 5.1.2, linting fails on code in that directory when not ignored.

I think this means that this was a bug and has since been fixed, since presumably linting would have failed had the files not been clobbered by Gradle when using the earlier version.

This sequence of Gradle commands results in a successful build and leaves the generated code, whereas before this would result in clobbered code:
```
./gradlew clean compileKotlin spotlessapply build
```

where `compileKotlin` is declared to depend on `openApiGenerate`.

Thanks for the guidance - always nice when an upgrade is the answer to my troubles. Should have tried that first myself.

**Comment by nedtwigg** [comment]
Thanks for opening, interesting to know about the old bug in 3.x
